src/glmtools.jl

```diff
@@ -499,6 +499,17 @@ false
 dispersion_parameter(D) = true
 dispersion_parameter(::Union{Bernoulli, Binomial, Poisson}) = false
 
+"""
+    _safer_int(x::T)
+    
+Convert to Int, when `x` is within 1 eps of an integer.
+"""
+function _safer_int(x::T) where {T<:Base.IEEEFloat}
+    r = round(Int, x)
+    abs(x - r) <= eps(x) && return r
+    throw(InexactError(:safer_int, T, x))
+end
```

Comment:
```suggestion
end

_safer_int(x) = Int(x)
```

I haven't combed through the GLM internals in a bit but I _believe_ this would be technically breaking without a fallback method, since `y * wt` may not necessarily be `Float16`, `Float32`, or `Float64`, and currently it works as long as its type has a method for the `Int` constructor.

Reply:
ah yeah good call